### ISSUE TITLE
[Paper Note] BitNet Distillation, Xun Wu+, arXiv'25, 2025.10

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#3324]
# URL
- https://arxiv.org/abs/2510.13998
# Authors
  - Xun Wu
  - Shaohan Huang
  - Wenhui Wang
  - Ting Song
  - Li Dong
  - Yan Xia
  - Furu Wei
# Abstract
  - In this paper, we present BitNet Distillation (BitDistill), a lightweight pipeline that fine-tunes off-the-shelf full-precision LLMs (e.g., Qwen) into 1.58-bit precision (i.e., ternary weights {-1, 0, 1}) for specific downstream tasks, achieving strong task-specific performance with minimal computational cost. Specifically, BitDistill incorporates three key techniques: the SubLN module, as introduced in BitNet; multi-head attention distillation, based on MiniLM; and continual pre-training, which serves as a crucial warm-up step to mitigate the scalability issue of the performance gap between finetuned full-precision and 1.58-bit LLMs on specific tasks. Experimental results show that BitDistill achieves performance comparable to the full-precision counterpart models across model size, while enabling up to 10x memory savings and 2.65x faster inference on CPUs. Code is available at https://github.com/microsoft/BitNet.
# Translation (by gpt-4o-mini)
- 本論文では、BitNet Distillation（BitDistill）を提案します。これは、オフ・ザ・シェルフのフル精度LLM（例：Qwen）を特定の下流タスク向けに1.58ビット精度（すなわち、三値重み{-1, 0, 1}）にファインチューニングする軽量なパイプラインであり、最小限の計算コストで強力なタスク特化型のパフォーマンスを達成します。具体的には、BitDistillは3つの重要な技術を組み込んでいます：BitNetで導入されたSubLNモジュール、MiniLMに基づくマルチヘッドアテンション蒸留、そして特定のタスクにおけるファインチューニングされたフル精度LLMと1.58ビットLLMのパフォーマンスギャップのスケーラビリティ問題を軽減するための重要なウォームアップステップとして機能する継続的な事前学習です。実験結果は、BitDistillがモデルサイズにわたってフル精度の対応モデルと同等のパフォーマンスを達成し、最大10倍のメモリ節約とCPU上での2.65倍の高速推論を可能にすることを示しています。コードはhttps://github.com/microsoft/BitNetで入手可能です。
# Summary (by gpt-4o-mini)
- BitNet Distillation（BitDistill）は、フル精度LLMを1.58ビット精度にファインチューニングする軽量なパイプラインで、計算コストを抑えつつ高いタスク特化型パフォーマンスを実現します。主な技術には、SubLNモジュール、MiniLMに基づくアテンション蒸留、継続的な事前学習が含まれ、これによりフル精度モデルと同等の性能を達成し、メモリを最大10倍節約し、CPU上での推論を2.65倍高速化します。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/_akhaliq/status/1979209909444001822?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
SubLN, MiniLMについては
- #1899
- #3345 

を参照のこと。

**Comment by AkihikoWatanabe** [comment]
既存LLMを特定タスクに1.58bitでSFTする際に、full-precisionと同等の性能を保つ方法を提案している研究。full-precision LLMを1.58 bitでSFTをするとfp16で学習した場合のbaselineと比較してパフォーマンスが大きく低下するが（そしてその傾向はモデルサイズが大きいほど強い）、提案手法を利用するとfp16でSFTした場合と同等の性能を保ちながら、inference-speed 2.65倍、メモリ消費量1/10になる模様。
<img width="1015" height="644" alt="Image" src="https://github.com/user-attachments/assets/cafa8ad5-7cce-4466-a208-07bb51dcd953" />

手法としては、3段階で構成されており
- Stage1: low-bitに量子化されたモデルではactivationの分散が大きくなり学習の不安定さにつながるため、アーキテクチャとしてSubLNを導入して安定化を図る
- Stage2: Stage1で新たにSubLNを追加するので事前学習コーパスの継続事前学習する
- Stage3: full-precisionでSFTしたモデルを教師、1.58-bitに量子化したモデルを生徒とし、logits distillation (input x, output yが与えられた時に教師・生徒間で出力トークンの分布のKL Divergenceを最小化する)、MiniLMで提案されているMHAのdistillation（q-q/k-k/v-vの内積によってsquaredなrelation mapをQ, K, Vごとに作成し、relation mapのKL Divergenceが教師・生徒間で最小となるように学習する）を実施する
- 最終的に `L_CE + \lambda L_LD + \ganma L_AD` を最小化する。ここで、L_CEはdownstream datasetに対するcross-entropy lossであり、L_LD, L_ADはそれぞれ、logit distillation, Attention Distillationのlossである。

**Comment by AkihikoWatanabe** [comment]
ポイント解説:https://x.com/jiqizhixin/status/1980968125547139259?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q
